### ISSUE TITLE
Printed pdf is not showing in dutch language

### ISSUE THREAD

**Issue by deepak-netwin** [fooman/printorderpdf-m2#7]
I tried changing the admin locale but still it's not working.
I've deployed content for that language and also cleared cache. But no luck.
Anything I'm missing?

**Comment by fooman** [comment]
@deepak-netwin have you tested the behaviour of the printed invoice?

**Comment by deepak-netwin** [comment]
@fooman Yes, In another ticket you replied that this extension takes admin locale and prints pdf in that language. However, I found that it doesn't work in that way too. It just prints the PDF in english.
That ticket was already closed so I've created this new one. Sorry for that.

**Comment by mbm-peter** [comment]
I have this same problem. I tried debugging the issue and I have found that I get the problem deep inside the call of getPaymentBlock(). At some point the Magento_Payment files are tried to be loaded with area emulation on (for "frontend"). This way it will to load the theme "frontend/Magento/backend", and the theme_dir won't get filled with the right value.

This is the error message on the invoice "Error filteren van template: Required parameter 'theme_dir' was not passed".

**Comment by fooman** [comment]
@mbm-peter just to confirm is the behaviour that you are seeing the same for the order confirmation pdf as well as for the invoice pdf?

Any detailed steps that you can provide on how we could replicate this would be great (including steps for installing the locale, etc).

I have come across this "frontend/Magento/backend" once but that was in relation to a deleted store view.

**Comment by mbm-peter** [comment]
In module-theme/Model/Theme.php#getArea if I disable the isAreaCodeEmulated() if check, I do get to see the invoice.

We haven't deleted a storeview.

**Comment by mbm-peter** [comment]
I now tried installing the module on a default Magento without changes. First order, and immediately I get this white order when printing from the admin. http://mous.es/vpgUEh.jpg
And invoice also doesn't work. Magento version 2.1.2

**Comment by fooman** [comment]
@mbm-peter thanks for the feedback. Just to confirm in your latest test you installed 2.1.7, then you installed Print Order Pdf and that's it? Or did you also install a dutch language pack? If yes which one and how do you install it.

Generally if there is an already issue with the default invoice Print Order Pdf will also not work since it is built on top of it - see [here](https://github.com/fooman/printorderpdf-m2/blob/develop/src/Model/Pdf/Order.php#L14).

We have a demo up [here](http://printorderpdf.demo-m2.fooman.co.nz/admin) - without the dutch language though.

**Comment by mbm-peter** [comment]
I have disabled the Fooman Pdf modules now and I don't get a problem when printing the invoice. When I reenable the module I get the error again.

I also tried to disable only PdfCustomiser en EmailAttachments and I get error for missing arguments in a construct call. In vendor/fooman/printorderpdf-m2/src/Model/Pdf/Order.php I had to add the arguments of the parent class to the __construct function to make this work. Then I can get both Order and Invoice pdf.

If I reenable PdfCustomiser I still get the themedir error.

I removed everything I could find for the dutch translation, all csv files, but that doesn't change the error.

**Comment by fooman** [comment]
@mbm-peter thanks for the update - apologies I did not realise that you are also using Pdf Customiser. Would you mind getting in touch via our support@fooman.co.nz email address. If you could include which language pack you installed that be great as that would help us in trying to reproduce this issue.
